src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int seed = 13;
+        seed += (length != null) ? seed * length : 0;
+        seed += (width != null) ? seed * width : 0;
+        seed += (color != null) ? seed * color.hashCode() : 0;
+        return seed;
+    }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle current = (Rectangle) rectangle;
+            return (width == current.getWidth() || width.equals(current.getWidth()))
+                    && (length == current.getLength() || length.equals(current.getLength()))
+                    && (color == current.getColor() || color.equals(current.getColor()));
```

Comment:
what if `width` is null and `current.width` is not null?

Reply:
done